src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/TouchBehavior.android.cs

```diff
@@ -0,0 +1,581 @@
+		&& this.view is not null
+		&& ripple.IsAlive()
+		&& this.view.IsAlive()
+		&& (IsAndroidVersionAtLeast(23) ? this.view.Foreground : this.view.Background) == ripple
+		&& element is MView view
+		&& view.GestureRecognizers.Any(gesture => gesture is TapGestureRecognizer);
+
+	/// <summary>
+	/// Attaches the behavior to the platform view.
+	/// </summary>
+	/// <param name="bindable">Maui Visual Element</param>
+	/// <param name="platformView">Native View</param>
+	protected override void OnAttachedTo(VisualElement bindable, AView platformView)
+	{
+		Element = bindable;
+		view = platformView;
+		viewGroup = Microsoft.Maui.Platform.ViewExtensions.GetParentOfType<ViewGroup>(platformView);
+		if (IsDisabled)
+		{
+			return;
+		}
+
+		platformView.Touch += OnTouch;
+		UpdateClickHandler();
+		accessibilityManager = platformView.Context?.GetSystemService(Context.AccessibilityService) as AccessibilityManager;
+
+		if (accessibilityManager is not null)
+		{
+			accessibilityListener = new AccessibilityListener(this);
+			accessibilityManager.AddAccessibilityStateChangeListener(accessibilityListener);
+			accessibilityManager.AddTouchExplorationStateChangeListener(accessibilityListener);
+		}
+
+		if (!IsAndroidVersionAtLeast((int)BuildVersionCodes.Lollipop) || !NativeAnimation)
+		{
+			return;
+		}
+
+		platformView.Clickable = true;
+		platformView.LongClickable = true;
+
+		CreateRipple();
+		ApplyRipple();
+
+		platformView.LayoutChange += OnLayoutChange;
+	}
+
+	/// <summary>
+	/// Detaches the behavior from the platform view.
+	/// </summary>
+	/// <param name="bindable">Maui Visual Element</param>
+	/// <param name="platformView">Native View</param>
+	protected override void OnDetachedFrom(VisualElement bindable, AView platformView)
+	{
+		element = bindable;
+		view = platformView;
+
+		if (element is null)
+		{
+			return;
+		}
+
+		try
+		{
+			if (accessibilityManager is not null && accessibilityListener is not null)
+			{
+				accessibilityManager.RemoveAccessibilityStateChangeListener(accessibilityListener);
+				accessibilityManager.RemoveTouchExplorationStateChangeListener(accessibilityListener);
+				accessibilityListener.Dispose();
+				accessibilityManager = null;
+				accessibilityListener = null;
+			}
+
+			RemoveRipple();
+
+			if (view is not null)
+			{
+				view.LayoutChange -= OnLayoutChange;
+				view.Touch -= OnTouch;
+				view.Click -= OnClick;
+			}
+
+			if (rippleView is not null)
+			{
+				rippleView.Pressed = false;
+				viewGroup?.RemoveView(rippleView);
+				rippleView.Dispose();
+				rippleView = null;
+			}
+		}
+		catch (ObjectDisposedException)
+		{
+			// Suppress exception
+			Trace.WriteLine("TouchBehavior is already disposed.");
+		}
+
+		isHoverSupported = false;
+	}
+
+	static ColorStateList GetColorStateList(MColor? color)
+	{
+		var animationColor = color;
+		animationColor ??= defaultNativeAnimationColor;
+
+		return new ColorStateList([Array.Empty<int>()], [animationColor.ToAndroid()]);
```

Comment:
let's type less xP
```suggestion
		return new ColorStateList([[]], [animationColor.ToAndroid()]);
```

Reply:
This isn't my code but I'd like to defend the practise of Array.Empty, its good code hygiene because only 1 empty array will be created in the app domain. Does `[]` work the same way?, if so ignore me... However as a rule I think more code should be using the `Array.Empty`/`Enumerable.Empty` paradigm!

Reply:
`[]` makes sure that compiler will use the best way to initialize that array, so under the hood it will not allocate, just like Array.Empty, as you can see [here](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATABgLABQ2AjIYRsQJwAUARIDwbgIPu0CUA3GQYQG4CGUAAhgBbAA4AXAJ4BBKIIC8A2VF6SAdAFExUgDwBLAHbiAfNXadD4gNoBdIdplyAwgMW22QA=)